c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+
+	float TemporalUserChoice;
+	std::cin >> TemporalUserChoice;
+
+	UserChoice = (CategoryManagerCommand)ValidateInputAndReturnValidatedInput(TemporalUserChoice, CategoryManagerMessage, 1, 3);
+
+	switch (UserChoice)
+	{
+	case CategoryManagerCommand::CreateCategory:
+		OpenCreateCategory();
+		break;
+
+	case CategoryManagerCommand::RemoveCategory:
+		OpenRemoveCategory();
+		break;
+
+	case CategoryManagerCommand::MainMenu:
+		ActiveCommand = MenuCommand::MainMenu;
+		return;
+
+	default:
+		std::cout << "CategoryManager Error, please contact support";
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenGameDisplayerMenu() {
+	system("cls");
+
+	if (NumberOfGames == 0)
+	{
+		std::cout << "There are currently no games to display. Please type Enter to return to the main menu. ";
+
+		std::cin.ignore(500, '\n');
+		if (std::cin.peek() == '\n')
+		{
+			OpenMainMenu();
+		}
+		return;
+	}
+
+	if (Uncategorized.GetCurrentNumberOfGames() != 0)
+	{
+		std::cout << "---------------------------------------------------------\n" << "Uncategorized Games\n" << std::endl;
+		for (int i = 0; i < Uncategorized.GetCurrentNumberOfGames(); i++)
+		{
+			FVideogame Game = Uncategorized.GetGame(i);
+			std::cout << Game.GetName() << "\t by " << Game.GetStudio() << "\t published on " << Game.GetFormattedDate() << std::endl;
+		}
+	}
+
+	if (!CategoryContainer.IsEmpty())
+	{
+		for (int i = 0; i < CategoryContainer.GetCurrentNumberOfCategories(); i++)
+		{
+			if (CategoryContainer.GetCategory(i).GetCurrentNumberOfGames() != 0)
+			{
+				std::cout << "---------------------------------------------------------\n" << CategoryContainer.GetCategory(i).GetName() << " Games\n" << std::endl;
+
+				for (int j = 0; j < CategoryContainer.GetCategory(i).GetCurrentNumberOfGames(); j++)
+				{
+					FVideogame Game = CategoryContainer.GetCategory(i).GetGame(j);
+					std::cout << Game.GetName() << "\t by " << Game.GetStudio() << "\t published on " << Game.GetFormattedDate() << std::endl;
+				}
+			}
+		}
+	}
+
+	std::cout << "\nPlease type Enter to return to the previous menu ";
+
+	std::cin.ignore(500, '\n');
+	if (std::cin.peek() == '\n')
+	{
+		OpenMainMenu();
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenCreateCategory()
+{
+	system("cls");
+
+	if (CategoryContainer.IsFull())
+	{
+		std::cout << "There is no available space right now to create another category, please type Enter to return to the previous menu";
+
+		std::cin.ignore(500, '\n');
+		if (std::cin.peek() == '\n')
+		{
+			OpenCategoryManagerMenu();
+		}
+		return;
+	}
+
+	std::cout << "Please enter a name for the new category: ";
+	std::string TemporaryCategoryName;
```

Comment:
you should condense the name `TemporaryCategoryName` to just `NewCategoryName` or `CategoryName`

Reply:
Same as before, it might not be the actual NewCategoryName, but just a first attempt given to us by the user.

Reply:
However I think I'm gonna' change the name to TemporalCategoryName to follow the example of the ones I created before.